docker/generate-config.js

```diff
@@ -1,68 +1,73 @@
 const fs = require("fs");
 const path = require("path");
 
-const configPath = path.resolve("/usr/share/nginx/html/config.js");
+const htmlPath =
+  process.env.NODE_ENV === "development"
+    ? path.resolve(__dirname, "..", "public")
+    : "/usr/share/nginx/html";
+const configPath = path.resolve(htmlPath, "config.js");
+const themesPath = path.resolve(htmlPath, "themes");
 
 console.log("Pre-start: Generating config file…");
 
-let content = "";
-try {
-  content = fs.readFileSync(configPath, { encoding: "utf-8" });
-} catch {
-  console.log(
-    `Pre-start: ⚠️ No existing config file found under "${configPath}".`
-  );
-}
-
-const convertToNativeType = (input) => {
-  if (["true", "false"].includes(input)) {
-    return input === "true";
-  }
+const config = readConfig();
 
-  if (!Number.isNaN(+input)) {
-    return +input;
+Object.entries(process.env).forEach(([key, value]) => {
+  if (key.startsWith("VOCASCAN_")) {
+    config[key.replace(/^VOCASCAN_/, "")] = convertToNativeType(value);
   }
+});
 
-  return input;
-};
+if (fs.existsSync(themesPath) && fs.lstatSync(themesPath).isDirectory()) {
+  const themes = Object.fromEntries(
+    fs
+      .readdirSync(themesPath)
+      .map((theme) => [
+        theme.replace(/(themes\/|.css)/g, ""),
+        `themes/${theme}`,
+      ])
+  );
+  config.themes = { ...config.themes, ...themes };
+}
 
-const envVars = Object.entries(process.env).reduce((acc, [key, value]) => {
-  if (key.startsWith("VOCASCAN_")) {
-    acc[key.replace(/^VOCASCAN_/, "")] = convertToNativeType(value);
-  }
+writeConfig(config);
 
-  return acc;
-}, {});
+console.log(
+  `Pre-start: ✓ Successfully written config file to "${configPath}".`
+);
 
-const existing = content.match(/window\.VOCASCAN_CONFIG *= *(\{[\s\S]*\})/);
-let existingConfig = {};
+function readConfig() {
+  if (!fs.existsSync(configPath)) {
+    console.warn(
+      `Pre-start: ⚠️ No existing config file found under "${configPath}".`
+    );
+    process.exit();
+  }
+  const content = fs.readFileSync(configPath, "utf-8");
+  const existing = content.match(
+    /window\.VOCASCAN_CONFIG\s*=\s*JSON.parse\(`([\s\S]*)`\)/
+  );
+  if (!existing || existing.length < 2) {
+    console.warn("Could not find existing config in config.js");
+    process.exit();
+  }
+  return JSON.parse(existing[1]);
+}
 
-if (existing && existing[1]) {
-  // evaluate to get the object string as object
-  // eslint-disable-next-line no-eval
-  existingConfig = eval(`(${existing[1]})`);
+function writeConfig(config) {
+  const json = JSON.stringify(config, null, 4);
+  const newContent = `window.VOCASCAN_CONFIG = JSON.parse(\`${json}\`);\n`;
+  fs.writeFileSync(configPath, newContent, "utf-8");
```

Comment:
can you add the error catching and success message back.

Reply:
Success message is still there. The error message is not neccessary, it will never be thrown

Reply:
But I think a custom error message looks better than a thrown one.

Reply:
If that's your argument, then we should wrap every function call in a try-catch block